src/globus_cli/commands/group/set_policies.py

```diff
@@ -0,0 +1,103 @@
+from typing import Any
+
+import click
+
+from globus_cli.login_manager import LoginManager
+from globus_cli.parsing import command
+from globus_cli.termio import formatted_print
+
+from ._common import group_id_arg
+
+
+@click.option(
+    "--high-assurance/--not-high-assurance",
+    help="Flag if the group will enforce high assurance access policies or not.",
+)
+@click.option(
+    "--authentication-timeout",
+    type=int,
+    help=(
+        "Time in seconds before a user must reauthenticate to access a "
+        "high assurance group"
+    ),
+)
+@click.option(
+    "--visibility",
+    type=click.Choice(("authenticated", "private"), case_sensitive=False),
+    help=(
+        "Determine who can see the group. "
+        "If authenticated, all authenticated users can see the group. "
+        "If private, only members and managers can see the group."
+    ),
+)
+@click.option(
+    "--member-visibility",
+    type=click.Choice(("members", "managers"), case_sensitive=False),
+    help=(
+        "Determine who can see members of the group. "
+        "If members, members can see all other members. "
+        "If managers, only managers can see members."
+    ),
+)
+@click.option(
+    "--join-requests/--no-join-requests",
+    help="Flag if request to join the group are allowed or not.",
+)
+@click.option(
+    "--signup-field",
+    type=click.Choice(
+        (
+            "institution",
+            "current_project_name",
+            "address",
+            "city",
+            "state",
+            "country",
+            "address1",
+            "address2",
+            "zip",
+            "phone",
+            "department",
+            "field_of_science",
+        ),
+        case_sensitive=False,
+    ),
+    multiple=True,
+    help=(
+        "Field required from users to apply for group membership. "
+        "Pass this option multiple times to require multiple fields. "
+        "Any existing required fields will be overwritten if given."
+    ),
```

Comment:
The behavior of this flag in particular is making me wonder if we want to pursue a suite of commands for managing a group's policies.

Do we actually want
```
globus groups policies set  # current implementation
globus groups policies signup-field add
globus groups policies signup-field remove
```
?

The group `policies` object _probably_ won't grow in complexity, but if it does, and there are more fields like `signup-field` which benefit from special-handling, it might be nice to have a dedicated namespace for such commands.

If we think that is the direction we want to take, the _only_ change I'd suggest today is to rename to `policies set`. Fancy `signup-field add/remove` support can be put on the roadmap.